src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -596,10 +610,16 @@
       },
       code: function(X) {
         var objToAdd = X.objectSummaryView ? X.objectSummaryView : X.summaryView;
+        var onExecuteFunc;
+
+        if ( X.controllerMode && X.controllerMode.equals(foam.u2.ControllerMode.SUMMARY) )
+          onExecuteFunc =  this.approveWithMemo.bind(this);
```

Comment:
You don't have to make this change, but since these are used as callbacks/listeners, if you moved them from methods: to listeners: then you wouldn't need to .bind(this) them, because listeners are pre-bound.

Reply:
fixed